pycontrails/models/cocip/cocip.py

```diff
@@ -1166,49 +1166,43 @@ def _bundle_results(self) -> None:
         # ---
         # Create contrail dataframe (self.contrail)
         # ---
-        dfs = [contrail.dataframe for contrail in self.contrail_list]
-        dfs = [df.assign(timestep=t_idx) for t_idx, df in enumerate(dfs)]
-        self.contrail = pd.concat(dfs)
+        for t_idx, contrail in enumerate(self.contrail_list):
+            contrail["timestep"] = np.full(contrail.size, t_idx)
+
+        self.contrail = GeoVectorDataset.sum(self.contrail_list).dataframe
```

Comment:
This change essentially replaces `pd.concat` with `np.concatenate`. I've found the later to be a bit faster but also use a lot less memory.